### PR TITLE
Feat/best streak

### LINKED ISSUE #85: Manage best streak
## Summary
Add a feature to save the best streak for a habit.

### PR DESCRIPTION
**Pull request by ShenolShengov**
## Summary
Add funcionality to save best sterak for habit

## Issue
Closes #85 

## Checklist
- [x] Code builds successfully
- [x] Documentation updated (if needed)
- [x] No breaking changes (or documented)
- [x] Code review requested

### CODE REVIEW

File: src/main/java/com/habittracker/api/auth/utils/AuthUtils.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public final class AuthUtils {
+
+  public static ZoneId getUserTimeZone() {
+    return parseTimeZone(getDetails().getProfile().getTimezone());
+  }
+
+  private static UserDetailsImpl getDetails() {
+    return (UserDetailsImpl) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```
Comment (hyuseinleshov) [inline src/main/java/com/habittracker/api/auth/utils/AuthUtils.java]: The method assumes authentication principal exists and is of type UserDetailsImpl. In edge cases (unauthenticated requests, test scenarios), this could throw ClassCastException or NullPointerException.

**Review by hyuseinleshov — CHANGES_REQUESTED** [review]
Overall it looks good
